base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/views/ActionsViewTable.java

```diff
@@ -0,0 +1,115 @@
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.TableColumn;
+
+import uk.ac.stfc.isis.ibex.scriptgenerator.ActionParameter;
+import uk.ac.stfc.isis.ibex.scriptgenerator.table.ScriptGeneratorAction;
+import uk.ac.stfc.isis.ibex.scriptgenerator.table.ActionsTable;
+import uk.ac.stfc.isis.ibex.ui.tables.ColumnComparator;
+import uk.ac.stfc.isis.ibex.ui.tables.DataboundCellLabelProvider;
+import uk.ac.stfc.isis.ibex.ui.tables.DataboundTable;
+import uk.ac.stfc.isis.ibex.ui.tables.NullComparator;
+import uk.ac.stfc.isis.ibex.ui.widgets.StringEditingSupport;
+
+/**
+ * A table that holds the properties for a target.
+ */
+@SuppressWarnings("checkstyle:magicnumber")
+public class ActionsViewTable extends DataboundTable<ScriptGeneratorAction> {
+
+    private ActionsTable actionsTable;
+	/**
+     * Default constructor for the table. Creates all the correct columns.
+     * 
+     * @param parent
+     *            The parent composite that this table belongs to.
+     * @param style
+     *            The SWT style of the composite that this creates.
+     * @param tableStyle
+     *            The SWT style of the table.
+     * @param scriptGeneratorTable 
+     */
+    public ActionsViewTable(Composite parent, int style, int tableStyle, ActionsTable actionsTable) {
+        super(parent, style, tableStyle | SWT.BORDER);
+        this.actionsTable = actionsTable;
+        initialise();
+    }
+
+    
+    @Override
+    protected void addColumns() {    	
+        for (ActionParameter actionParameter:this.actionsTable.getActionParameters()) {
+        	addColumn(actionParameter);
+        }
+    }
+
+	private void addColumn(ActionParameter actionParameter) {
+		String columnName = actionParameter.getName();
+		TableViewerColumn column = createColumn(
+				columnName, 
+				2,
+				new DataboundCellLabelProvider<ScriptGeneratorAction>(observeProperty(columnName)) {
+					@Override
+					protected String stringFromRow(ScriptGeneratorAction row) {
+						// TODO Auto-generated method stub
+						return row.getActionParameterValue(columnName);
+					}
+					
+				});
+		
+        column.setEditingSupport(new StringEditingSupport<ScriptGeneratorAction>(viewer(), ScriptGeneratorAction.class) {
+
+            @Override
+            protected String valueFromRow(ScriptGeneratorAction row) {
+                return row.getActionParameterValue(columnName);
+            }
+
+            @Override
+            protected void setValueForRow(ScriptGeneratorAction row, String value) {
+                row.setActionParameterValue(columnName, value);
+            }
+        });	
+		
+	}
+
+	protected SelectionAdapter getColumnSelectionAdapter(final TableColumn column, final int index) {
+        SelectionAdapter selectionAdapter = new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+            }
+        };
+        return selectionAdapter;
+    }
```

Comment:
Documentation on why we're doing this would be good.

Reply:
This method no longer useful, it was a first attempt at stopping the columns from being reordered. This is done by overriding `ColumnComparator`.

Reply:
Alistair has removed it